scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1373,8 +1373,28 @@
     (when-not (synthetic-pod? pod-name)
       (let [[resolved-config reason] (config-incremental/resolve-incremental-config task-id :add-finalizer "false")]
         (if (= "true" resolved-config)
-          (.setFinalizers metadata (list FinalizerHelper/collectResultsFinalizer)))))(when pod-annotations
-                                                                                       (.setAnnotations metadata pod-annotations))
+          (.setFinalizers metadata (list FinalizerHelper/collectResultsFinalizer)))))
+    (let [pod-annotations'
+          (if (synthetic-pod? pod-name)
+            pod-annotations
+            ; For non-synthetic (real job) pods, when configured to do so,
+            ; we add a pod annotation indicating that Kubernetes should
+            ; use all of the group IDs the user is a member of, in order
+            ; to avoid contradictions between which group Cook thinks a
+            ; user belongs to and which group Kubernetes thinks the user
+            ; belongs to.
+            (let [[resolved-config _]
+                  (config-incremental/resolve-incremental-config
+                    task-id :add-use-all-gids-annotation "false")
```

Comment:
you need a java.util.UUID here and you want the job uuid not instance uuid

Reply:
Above for the `:add-finalizer` one we're using `task-id`

Reply:
yeah, that's wrong too. was approved in PR 1954